Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1982,30 +1982,77 @@ function Remove-FilePermissions
   $removePermissionsAcl.RemoveAccessRule($accessRule)
   $removePermissionsAcl | Set-Acl $filePath
 }
-    
+
 function New-IdentityEncryptionCertificate {
     param(
         [string] $subject = "$env:computername.$((Get-WmiObject Win32_ComputerSystem).Domain.tolower())",
-        [string] $certStoreLocation = "Cert:\LocalMachine\My"
+        [string] $certStoreLocation = "Cert:\LocalMachine\My",
+        [string] $friendlyName = "Fabric Identity Signing Encryption Certificate"
     )
-    $cert = New-SelfSignedCertificate -Type Custom -KeySpec None -Subject $subject -KeyUsage DataEncipherment -KeyAlgorithm RSA -KeyLength 2048 -CertStoreLocation $certStoreLocation
+    $cert = New-SelfSignedCertificate `
+        -Type Custom `
+        -KeySpec None `
+        -Subject $subject `
+        -KeyUsage DataEncipherment `
+        -KeyAlgorithm RSA `
+        -KeyLength 2048 `
+        -CertStoreLocation $certStoreLocation `
+        -FriendlyName $friendlyName
+
     return $cert
 }
 
 function Test-IdentityEncryptionCertificateValid {
     param(
         [System.Security.Cryptography.X509Certificates.X509Certificate2] $encryptionCertificate
     )
+    # Update a month before
     $today = Get-Date
     return $encryptionCertificate.NotAfter -gt $today
 }
 
 function Remove-IdentityEncryptionCertificate {
     param(
-        [string] $encryptionCertificateThumbprint
+        [string] $encryptionCertificateThumbprint,
+        [string] $friendlyName = "Fabric Identity Signing Encryption Certificate"
     )
-    $cert = Get-Certificate $encryptionCertificateThumbprint
```

Comment:
Fun note: Authorization uses this function as well, this causes the Authorization install.config to have a signing/encryption certificate in it's own install.config section.

Authorization will need to be updated to either not use this, or to grab the encryption certificate value from the common section of the install.config

Reply:
I like having the common stuff like encryption certificate in common section.